// react-native-bottom-tabs/lib/commonjs/TabView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _BottomTabBarHeightContext = require("./utils/BottomTabBarHeightContext");
var _TabViewNativeComponent = _interopRequireDefault(require("./TabViewNativeComponent"));
var _useLatestCallback = _interopRequireDefault(require("use-latest-callback"));
var _DelayedFreeze = _interopRequireDefault(require("./DelayedFreeze"));
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
//@ts-ignore

const isAppleSymbol = icon => icon?.sfSymbol;
const ANDROID_MAX_TABS = 6;
const TabView = ({
  navigationState,
  renderScene,
  onIndexChange,
  onTabLongPress,
  getBadge,
  rippleColor,
  tabBarActiveTintColor: activeTintColor,
  tabBarInactiveTintColor: inactiveTintColor,
  getLazy = ({
    route
  }) => route.lazy,
  getLabelText = ({
    route
  }) => route.title,
  getIcon = ({
    route,
    focused
  }) => route.unfocusedIcon ? focused ? route.focusedIcon : route.unfocusedIcon : route.focusedIcon,
  getHidden = ({
    route
  }) => route.hidden,
  getActiveTintColor = ({
    route
  }) => route.activeTintColor,
  getTestID = ({
    route
  }) => route.testID,
  hapticFeedbackEnabled = false,
  // Android's native behavior is to show labels when there are less than 4 tabs. We leave it as undefined to use the platform default behavior.
  labeled = _reactNative.Platform.OS !== 'android' ? true : undefined,
  getFreezeOnBlur = ({
    route
  }) => route.freezeOnBlur,
  tabBar: renderCustomTabBar,
  tabBarStyle,
  tabLabelStyle,
  ...props
}) => {
  // @ts-ignore
  const focusedKey = navigationState.routes[navigationState.index].key;
  const [tabBarHeight, setTabBarHeight] = _react.default.useState(0);
  const [measuredDimensions, setMeasuredDimensions] = _react.default.useState();
  const trimmedRoutes = _react.default.useMemo(() => {
    if (_reactNative.Platform.OS === 'android' && navigationState.routes.length > ANDROID_MAX_TABS) {
      console.warn(`TabView only supports up to ${ANDROID_MAX_TABS} tabs on Android`);
      return navigationState.routes.slice(0, ANDROID_MAX_TABS);
    }
    return navigationState.routes;
  }, [navigationState.routes]);

  /**
   * List of loaded tabs, tabs will be loaded when navigated to.
   */
  const [loaded, setLoaded] = _react.default.useState([focusedKey]);
  if (!loaded.includes(focusedKey)) {
    // Set the current tab to be loaded if it was not loaded before
    setLoaded(loaded => [...loaded, focusedKey]);
  }
  const icons = _react.default.useMemo(() => trimmedRoutes.map(route => getIcon({
    route,
    focused: route.key === focusedKey
  })), [focusedKey, getIcon, trimmedRoutes]);
  const items = _react.default.useMemo(() => trimmedRoutes.map((route, index) => {
    const icon = icons[index];
    const isSfSymbol = isAppleSymbol(icon);
    if (_reactNative.Platform.OS === 'android' && isSfSymbol) {
      console.warn('SF Symbols are not supported on Android. Use require() or pass uri to load an image instead.');
    }
    return {
      key: route.key,
      title: getLabelText({
        route
      }) ?? route.key,
      sfSymbol: isSfSymbol ? icon.sfSymbol : undefined,
      badge: getBadge?.({
        route
      }),
      activeTintColor: (0, _reactNative.processColor)(getActiveTintColor({
        route
      })),
      hidden: getHidden?.({
        route
      }),
      testID: getTestID?.({
        route
      })
    };
  }), [trimmedRoutes, icons, getLabelText, getBadge, getActiveTintColor, getHidden, getTestID]);
  const resolvedIconAssets = _react.default.useMemo(() =>
  // Pass empty object for icons that are not provided to avoid index mismatch on native side.
  icons.map(icon => icon && !isAppleSymbol(icon) ? _reactNative.Image.resolveAssetSource(icon) : {
    uri: ''
  }), [icons]);
  const jumpTo = (0, _useLatestCallback.default)(key => {
    const index = trimmedRoutes.findIndex(route => route.key === key);
    onIndexChange(index);
  });
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_BottomTabBarHeightContext.BottomTabBarHeightContext.Provider, {
    value: tabBarHeight,
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_TabViewNativeComponent.default, {
      ...props,
      ...tabLabelStyle,
      style: styles.fullWidth,
      items: items
      // When rendering a custom tab bar, icons can be React elements, which will not be properly resolved.
      ,
      icons: renderCustomTabBar ? undefined : resolvedIconAssets,
      selectedPage: focusedKey,
      tabBarHidden: !!renderCustomTabBar,
      onTabLongPress: ({
        nativeEvent: {
          key
        }
      }) => {
        const index = trimmedRoutes.findIndex(route => route.key === key);
        onTabLongPress?.(index);
      },
      onPageSelected: ({
        nativeEvent: {
          key
        }
      }) => {
        jumpTo(key);
      },
      onTabBarMeasured: ({
        nativeEvent: {
          height
        }
      }) => {
        setTabBarHeight(height);
      },
      onNativeLayout: ({
        nativeEvent: {
          width,
          height
        }
      }) => {
        setMeasuredDimensions({
          width,
          height
        });
      },
      hapticFeedbackEnabled: hapticFeedbackEnabled,
      activeTintColor: activeTintColor,
      inactiveTintColor: inactiveTintColor,
      barTintColor: tabBarStyle?.backgroundColor,
      rippleColor: rippleColor,
      labeled: labeled,
      children: trimmedRoutes.map(route => {
        if (getLazy({
          route
        }) !== false && !loaded.includes(route.key)) {
          // Don't render a screen if we've never navigated to it
          return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
            collapsable: false,
            style: styles.fullWidth
          }, route.key);
        }
        const focused = route.key === focusedKey;
        const freeze = !focused ? getFreezeOnBlur({
          route
        }) : false;
        return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
          style: [styles.screen, renderCustomTabBar ? styles.fullWidth : measuredDimensions],
          collapsable: false,
          pointerEvents: focused ? 'auto' : 'none',
          accessibilityElementsHidden: !focused,
          importantForAccessibility: focused ? 'auto' : 'no-hide-descendants',
          children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_DelayedFreeze.default, {
            freeze: !!freeze,
            children: renderScene({
              route,
              jumpTo
            })
          })
        }, route.key);
      })
    }), renderCustomTabBar?.() ?? null]
  });
};
const styles = _reactNative.StyleSheet.create({
  fullWidth: {
    width: '100%',
    height: '100%',
    flex: 1
  },
  screen: {
    position: 'absolute'
  }
});
var _default = exports.default = TabView;
