// react-native-bottom-tabs/lib/commonjs/TabViewNativeComponent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.__INTERNAL_VIEW_CONFIG = void 0;
var _codegenNativeComponent = _interopRequireDefault(require("react-native/Libraries/Utilities/codegenNativeComponent"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
//@ts-ignore

const NativeComponentRegistry = require('react-native/Libraries/NativeComponent/NativeComponentRegistry');
const {
  ConditionallyIgnoredEventHandlers
} = require('react-native/Libraries/NativeComponent/ViewConfigIgnore');
let nativeComponentName = 'RNCTabView';
const __INTERNAL_VIEW_CONFIG = exports.__INTERNAL_VIEW_CONFIG = {
  uiViewClassName: 'RNCTabView',
  directEventTypes: {
    topPageSelected: {
      registrationName: 'onPageSelected'
    },
    topTabLongPress: {
      registrationName: 'onTabLongPress'
    },
    topTabBarMeasured: {
      registrationName: 'onTabBarMeasured'
    },
    topNativeLayout: {
      registrationName: 'onNativeLayout'
    }
  },
  validAttributes: {
    items: true,
    selectedPage: true,
    icons: true,
    tabBarHidden: true,
    labeled: true,
    sidebarAdaptable: true,
    scrollEdgeAppearance: true,
    barTintColor: {
      process: require('react-native/Libraries/StyleSheet/processColor').default
    },
    translucent: true,
    rippleColor: {
      process: require('react-native/Libraries/StyleSheet/processColor').default
    },
    activeTintColor: {
      process: require('react-native/Libraries/StyleSheet/processColor').default
    },
    inactiveTintColor: {
      process: require('react-native/Libraries/StyleSheet/processColor').default
    },
    ignoresTopSafeArea: true,
    disablePageAnimations: true,
    activeIndicatorColor: {
      process: require('react-native/Libraries/StyleSheet/processColor').default
    },
    hapticFeedbackEnabled: true,
    fontFamily: true,
    fontWeight: true,
    fontSize: true,
    ...ConditionallyIgnoredEventHandlers({
      onPageSelected: true,
      onTabLongPress: true,
      onTabBarMeasured: true,
      onNativeLayout: true
    })
  }
};
var _default = exports.default = NativeComponentRegistry.get(nativeComponentName, () => __INTERNAL_VIEW_CONFIG);
